### PR TITLE
Only run once - Fixes #17 - Fixes #23

### LINKED ISSUE #17: Slow command running with isolated: true
With both servers started locally, the request time is about 2 seconds

```bash
127.0.0.1 - - [02/Jan/2018:16:08:10 -0300] "POST /test HTTP/1.1" 200 1273 2.0182
127.0.0.1 - - [02/Jan/2018:16:08:44 -0300] "POST /test HTTP/1.1" 200 1273 1.9583
127.0.0.1 - - [02/Jan/2018:16:10:36 -0300] "POST /test HTTP/1.1" 200 1273 2.6443
127.0.0.1 - - [02/Jan/2018:16:10:39 -0300] "POST /test HTTP/1.1" 200 1273 1.8920
127.0.0.1 - - [02/Jan/2018:16:10:42 -0300] "POST /test HTTP/1.1" 200 1273 1.9624
127.0.0.1 - - [02/Jan/2018:16:10:45 -0300] "POST /test HTTP/1.1" 200 1273 2.0506
```

### LINKED ISSUE #23: Test names not rendering
screenshot2

https://github.com/mumuki/mumuki-gobstones-runner/blob/master/lib/gobstones/batch_parser.rb#L23
^ A big `// TODO` that I can't figure out yet.

I'm not sure of what should I do to make it show the titles. They're in `example[:title]`, but the guy who renders the green tick is ignoring them.

Test file: https://github.com/sagrado-corazon-alcal/mumuki-guia-fundamentos-funciones/blob/ca9aa8d11bf2fb700a72a76941929bf8e3454163/00001_Y%20esto%2C%20%C2%BFcon%20qu%C3%A9%20se%20come%3F/test.yml

### PR DESCRIPTION
**Pull request by rodri042**
It did improved the execution speed.

Before:
screenshot3

After:
screenshot4

### CODE REVIEW

File: lib/gobstones/batch_parser.rb [hunk 1]
```diff
@@ -32,7 +32,7 @@ def parse_example(options, example)
 
       return_value = example[:postconditions][:return]
       if return_value
-        example[:title] = "#{options[:subject]}() -> #{return_value}" unless example[:title]
+        example[:name] = "#{options[:subject]}() -> #{return_value}" unless example[:name]
```
Comment (flbulgarelli) [inline lib/gobstones/batch_parser.rb]: @rodri042 was it `name` actually?

File: lib/test_hook.rb [hunk 2]
```diff
@@ -1,30 +1,11 @@
...
-  end
-
-  def compile_file_content(request)
-    @batch = Gobstones::BatchParser.parse request
-    @batch.to_json
-  end
-
-  def post_process_file(_file, result, status)
-    output = result.parse_as_json
+class GobstonesTestHook < Mumukit::Defaults::PrecompileHook
```
Comment (flbulgarelli) [inline lib/test_hook.rb]: Please inherit from `Mumukit::Defaults::TestHook`

**Comment by flbulgarelli** [comment]
Really nice! And the code of the hooks is now even smaller. 


Still we have to do some work on improving the speed. Could you discover which gems are causing the problem?

**Comment by luchotc** [comment]
Testing a bit i found this mulang error in rails console:

```
mulang: "Error in $.sample.ast: expected Object, encountered Null"
CallStack (from HasCallStack):
  error, called at src/Control/Fallible.hs:14:26 in mulang-3.6.1-IbajWLO153kHM6RoZGhMp2:Control.Fallible
```

File: lib/gobstones/batch_parser.rb (hunk 1 above)
Comment (rodri042) [reply lib/gobstones/batch_parser.rb]: Yes, Metatest expects `name`, but in the examples YAML it's `title`

**Comment by rodri042** [comment]
> Still we have to do some work on improving the speed. Did you discover which gems are causing the problem?

Nope. I just did some debugging and the `run!` method of the precompile hook takes almost all the response time. **Only when Docker is used**.

With `isolated: true`:
screenshot3

With `isolated: false`:
screenshot2

**Pull request merged**